shared/actions/platform-specific.native.js

```diff
@@ -56,7 +56,25 @@ function saveAttachmentDialog(filePath: string): Promise<NextURI> {
   return Promise.resolve(filePath)
 }
 
-function displayNewMessageNotification(text: string, convID: string, badgeCount: number) {
+function displayNewMessageNotification(text: string, convID: string, badgeCount: number, myMsgID: number) {
+  // Dismiss any non-plaintext notifications for the same message ID
+  if (isIOS) {
+    PushNotificationIOS.getDeliveredNotifications(param => {
+      let idents = []
```

Comment:
this could just be
```
PushNotificationIOS.getDeliveredNotifications(param => {
    PushNotificationIOS.removeDeliveredNotifications(params.filter(p => p.userInfo && p.userInfo.msgID === msgID).map(p => p.identifier))
})

```

Reply:
Sounds good.